lib/src/util/condition_scope_visitor.dart

```diff
@@ -255,6 +267,20 @@ abstract class ConditionScopeVisitor extends RecursiveAstVisitor {
   }
 
   @override
+  visitSwitchCase(SwitchCase node) {
+    node.visitChildren(this);
+    AstNode previousCase = node.parent.childEntities
+        .where((e) => e is AstNode)
+        .lastWhere((n) => n != node, orElse: () => null);
```

Comment:
It seems to me that
```
[0, 1, 2, 3, 4].lastWhere((e) => e != i)
```
will return `4` for every value of `i` other than `4`. I don't think this code works at all like you want it to, despite the test passing. (The test is incomplete; you need some cases that don't end in a break statement.)